cmake/compiler.cmake

```diff
@@ -110,6 +110,13 @@ set (CMAKE_CXX_FLAGS_RELWITHDEBINFO
 
 unset(CC_DEBUG_OPT)
 
+# Enable emission of the DWARF CFI (Call Frame Information) directives to the
+# assembler. This option is enabled by default on most compilers, but on GCC 7
+# for AArch64 and older it wasn't, so turn it on explicitly. When enabled, the
+# compiler emits .cfi_* directives that are required for the stack unwinding,
+# and defines __GCC_HAVE_DWARF2_CFI_ASM, which is checked below.
+add_compile_flags("C;CXX" "-fasynchronous-unwind-tables")
```

Comment:
I would rather add this flag here, instead of -funwind-tables:
https://github.com/tarantool/tarantool/blob/05002edf80d44a4a82da8be0d5e7bcda5349d47a/cmake/compiler.cmake#L161-L169

Reply:
This won't work because `add_compile_flags` must be called before `check_c_source_compiles`.

`-fexceptions` probably can be removed, I'm not sure.

Reply:
> -fexceptions probably can be removed, I'm not sure.

Nope, it can't be removed, since it is disabled by default for C:
> If you do not specify this option, GCC enables it by default for languages like C++ that normally require exception handling, and disables it for languages like C that do not normally require it.[^1]

[^1]: https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#Code-Gen-Options

Reply:
> This won't work because `add_compile_flags` must be called before `check_c_source_compiles`.

How about moving this check after adding necessary compile flags?

Reply:
Oops, I meant «`-funwind-tables` probably can be removed».

Reply:
> How about moving this check after adding necessary compile flags?

Done.